### ISSUE TITLE
Functions should be able to take function parameters

### ISSUE THREAD

**Issue by gwhitney** [josdejong/mathjs#2441]
Currently: (in the demo REPL on the home page)
```
nd(func,t) = (func(t+1e-10)-func(t))/1e-10
   nd(func, t)

nd(exp,2)
  Error: Undefined function func
```
[this came up in investigating PR #2392.]

But given that function objects can be created by function assignment, there doesn't seem to be any particular reason why in evaluation of nd, `func` can't be bound to the meaning of `exp` so that this evaluation will work. If there is interest in adding this capability to mathjs, I would be happy to work on a PR (although possibly we will run into some closure/funargs issues -- but since programming languages solve those, I should think mathjs can as well).

**Comment by gwhitney** [comment]
Turned out there isn't a funarg "problem" in that all mathjs variables are dynamic already, for better or worse. So this was a pretty simple change; if you think the extension is OK, as you can see there is a small PR that adds it. And if that _is_ merged, I think the simplicity of adding a numerical derivative function within the parser as per the example I added in the documentation adds another argument against merging #2392.

**Comment by josdejong** [comment]
Fix published in `v10.3.0`
